service/iot-service-samples/azure-sas-credential-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/AzureSasCredentialSample.java

```diff
@@ -0,0 +1,362 @@
+            System.exit(-1);
+        }
+
+        try
+        {
+            // FileUploadNotificationReceiver will use the same authentication mechanism that the ServiceClient itself uses,
+            // so the below APIs are also RBAC authenticated.
+            FileUploadNotificationReceiver fileUploadNotificationReceiver = serviceClient.getFileUploadNotificationReceiver();
+
+            log.info("Opening file upload notification receiver and listening for file upload notifications");
+            fileUploadNotificationReceiver.open();
+            FileUploadNotification fileUploadNotification = fileUploadNotificationReceiver.receive(FILE_UPLOAD_NOTIFICATION_LISTEN_SECONDS);
+
+            if (fileUploadNotification != null)
+            {
+                log.info("File upload notification received for device {}\n", fileUploadNotification.getDeviceId());
+            }
+            else
+            {
+                log.info("No feedback records were received");
+            }
+
+            fileUploadNotificationReceiver.close();
+        }
+        catch (IOException | InterruptedException e)
+        {
+            log.error("Failed to listen for file upload notification messages", e);
+            System.exit(-1);
+        }
+    }
+
+    private static void runJobClientSample(String iotHubHostName, AzureSasCredential credential)
+    {
+        // JobClient has some configurable options for HTTP read and connect timeouts, as well as for setting proxies.
+        // For this sample, the default options will be used though.
+        JobClientOptions options = JobClientOptions.builder().build();
+
+        // This constructor takes in your implementation of AzureSasCredential which allows you to use symmetric key based
+        // authentication without giving the client your connection string.
+        JobClient jobClient = new JobClient(iotHubHostName, credential, options);
+
+        try
+        {
+            log.info("Querying all active jobs for your IoT Hub");
+            Query deviceJobQuery = jobClient.queryDeviceJob(SqlQuery.createSqlQuery("*", SqlQuery.FromType.JOBS, null, null).getQuery());
+            int queriedJobCount = 0;
+            while (jobClient.hasNextJob(deviceJobQuery))
+            {
+                queriedJobCount++;
+                JobResult job = jobClient.getNextJob(deviceJobQuery);
+                log.info("Job {} of type {} has status {}", job.getJobId(), job.getJobType(), job.getJobStatus());
+            }
+
+            if (queriedJobCount == 0)
+            {
+                log.info("No active jobs found for your IoT Hub");
+            }
+        }
+        catch (IotHubException | IOException e)
+        {
+            log.error("Failed to query the jobs for your IoT Hub", e);
+            System.exit(-1);
+        }
+    }
+
+    private static void runDeviceMethodClientSample(String iotHubHostName, AzureSasCredential credential, String deviceId)
```

Comment:
I know we want to show usage of all clients here. But I feel the samples are really long this way. We should brainstorm about this a little bit. 1) Do we really need to call all clients as the only thing we want to show is how to construct and pass these credentials 2) If we want to add samples for all clients, should we make it as an option in the existing samples. That way customers can see there are two ways to authenticate based on the APIs they want to use rather than come to a sample specific to token auth.

Reply:
I agree that these samples are a bit big when all they really need to show off is the constructors. I'm not sure I like making the existing samples configurable to use two different authentication types though. That would be a lot of work, and I don't think it would help the samples stay focused. 

We could potentially split the samples I created in this PR into 1 sample per client to keep them brief. I feel there would be a lot of duplicate code to make this happen, so I'm a bit hesitant to do that.

We could also just remove the getTwin/invokeMethod/createDevice calls so that the sample simply calls the new constructors. I like that style for users who are just reading the samples, but when we run the sample, there would be no validation that the user actually provided the correct credentials until we actually make a service API call like getTwin/invokeMethod/etc.

Reply:
I agree we need to show some API call. What if you skip all clients and just show the sample with one? I feel very lengthy samples get confusing and is usually not recommended.